lib/google/ads/google_ads/google_ads_client.rb

```diff
@@ -116,80 +116,15 @@ def configure(&block)
-          # below.
-          logger = @logger
-
-          headers = {
-            :"developer-token" => @config.developer_token
-          }
-          if @config.login_customer_id
-            begin
-              login_customer_id = Integer(@config.login_customer_id)
-            rescue ArgumentError => e
-              if e.message.start_with?("invalid value for Integer")
-                raise ArgumentError.new("Invalid value for login_customer_id, must be integer")
-              end
-            end
-            if login_customer_id <= 0 || login_customer_id > 9_999_999_999
-              raise ArgumentError.new(
-                "Invalid login_customer_id. Must be an integer " \
-                "0 < x <= 9,999,999,999. Got #{login_customer_id}"
-              )
-            end
-            headers[:"login-customer-id"] = login_customer_id.to_s  # header values must be strings
-          end
-
-          if logger
-            logging_interceptor = Google::Ads::GoogleAds::LoggingInterceptor.new(logger)
-          end
-
-          if name.nil?
-            services = Factories.at_version(version).services.new(
-              service_path: service_path,
-              logging_interceptor: logging_interceptor,
-              credentials: get_updater_proc,
-              metadata: headers,
-              exception_transformer: ERROR_TRANSFORMER
-            )
-
-            patch_delegator = Class.new do
-              def initialize(services, headers, patch_callable)
-                @services = services
-                @headers = headers
-                @patch_callable = patch_callable
-              end
-
-              def respond_to_missing?(sym, include_private=false)
-                @services.respond_to?(sym, include_private)
-              end
-
-              def method_missing(name, *args)
-                @services.public_send(name, *args) do |cls|
-                  @patch_callable.call(cls, @headers)
-                  cls
-                end
-              end
-            end
-            patch_delegator.new(services, headers, method(:patch_lro_headers))
-          else
-            class_to_return = lookup_util.raw_service(name, version)
-            class_to_return = Class.new(class_to_return) do
-              unless service_path.nil? || service_path.empty?
-                const_set('SERVICE_ADDRESS', service_path.freeze)
-              end
-
-              const_set('GRPC_INTERCEPTORS', [logging_interceptor].compact)
-            end
-
-            patch_lro_headers(class_to_return, headers)
-
-            class_to_return.new(
-              credentials: get_updater_proc,
-              metadata: headers,
-              exception_transformer: ERROR_TRANSFORMER
-            )
-          end
+          LookupService.new(
+            name,
+            version,
+            lookup_util,
+            service_path,
+            @logger,
+            @config,
+            get_updater_proc
```

Comment:
This should be "get_credentials". It's actually a bug that it was using get_updater_proc and ignoring the @config.authentication. I came across this while working on service accounts.

Reply:
so this PR shouldn't change any behavior vs what's on top of master (that is we should not be conflating bug fixes and refactoring in PRs). I'd rather keep the current behavior (even if it's wrong, but passing the tests) and then do a second PR, on top of this one, to fix that behavior.

Reply:
Ok, that works for me.